types/sinon/index.d.ts

```diff
@@ -523,7 +524,7 @@ declare namespace Sinon {
          * Makes the stub call the provided @param func when invoked.
          * @param func
          */
-        callsFake(func: (...args: U) => V): SinonStub<U, V>;
+        callsFake(func: () => V): SinonStub<U, V>;
```

Comment:
@johnjesse Where is your source on this? The sinon source code, as far as I can tell (https://github.com/sinonjs/sinon/blob/master/lib/sinon/default-behaviors.js#L32-L34), allows a function with any amount of parameters to be passed in, simulating what would be called in the actual function. For example if you are stubbing an event emitter, you can have it call a fake function which takes in the entry being emitted as well as a call back:

```js
_emitter.expects('openReadStream').callsFake((_entry, callback) => {
  callback(null, readStream);
  readStream.emit('data', Buffer.from('Hello'));
  readStream.emit('data', Buffer.from('World'));
  readStream.emit('end');
});
```

Now I cant do this without some hacking around to by-pass the types. Clarification on this, or a revert, would be appreciated, thanks! :)

Reply:
I think you're right, I'll revert that commit and add a test case.

Reply:
Aweome! Thank you so much, and many thanks for improving the sinon types! <3

Reply:
PR to fix is in, #30656